config/config.go

```diff
@@ -311,7 +317,9 @@ func createConfigFromMetadata(md metadataJSON) *config {
 		c.svcEndpoint = md.Project.Attributes.OSConfigEndpointOld
 	}
 
-	return c
+	// Example instanceZone: projects/1071514831017/zones/us-west1-b
+	parts := strings.Split(c.instanceZone, "/")
+	c.svcEndpoint = fmt.Sprintf("%s-%s", parts[len(parts)-1], c.svcEndpoint)
```

Comment:
I like that this makes it easy to add the zone when someone overrides the endpoint, but what if someone wants to test against a non-zonal endpoint (eg. the existing global endpoint or a dev environment)?

Reply:
I think if the flag is set we should honor it, and only prepend the zone if the default is being used

Reply:
adding relevant offline dicussion for others


Ryan: I think respecting the override completely might also make things like testing staging difficult.
Maybe we could allow for a wild card (eg. `{zone}-staging.osconfig.sandbox.googleapis.com` vs `staging.osconfig.sandbox.googleapis.com`)

Andrew Jackura, 16 mins
yeah, hat would work
better than my idea of anther option to set
we should just do a string replace
so if {zone} is not set its just a noop